repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3481,6 +3482,77 @@ public void deleteClassificationOnlyPropagation(Set<String> deletedEdgeIds) thro
         }
     }
 
+    public void deleteClassificationOnlyPropagation(String deletedEdgeId) throws AtlasBaseException {
+        RequestContext.get().getDeletedEdgesIds().clear();
+        RequestContext.get().getDeletedEdgesIds().add(deletedEdgeId);
+
+        AtlasEdge edge = graph.getEdge(deletedEdgeId);
+
+        boolean isRelationshipEdge = deleteDelegate.getHandler().isRelationshipEdge(edge);
+        boolean isTermEntityEdge = isTermEntityEdge(edge);
+
+        if (edge == null || !isRelationshipEdge) {
+            return;
+        }
+
+        List<AtlasVertex> currentClassificationVertices = getPropagatableClassifications(edge);
```

Comment:
We are passing classification Vertex then why we are extracting it again for edge

Reply:
If the task is created with edge and classification then it should be executed with those only. Here we are again fetching classifications and and looping through it. This is nullifying the benefit of breaking the task into parts. We could have just pass task with single edge.

Reply:
Thanks for the heads up. I've addressed your comments and I created another PR for staging. Once approved I will test there.